app/helpers/react_components/mentoring/session.rb

```diff
@@ -51,16 +52,15 @@ def links
 
       def iterations
         if discussion
-          comment_counts = ::Mentor::DiscussionPost.
-            where(discussion: discussion).
+          comment_counts = discussion.posts.
             group(:iteration_id, :seen_by_mentor).
             count
         end
 
         solution.iterations.map do |iteration|
           counts = discussion ? comment_counts.select { |(it_id, _), _| it_id == iteration.id } : nil
           num_comments = discussion ? counts.sum(&:second) : 0
-          unread = discussion ? counts.reject { |(_, seen), _| seen }.present? : 0
+          unread = discussion ? counts.reject { |(_, seen), _| seen }.present? : false
```

Comment:
Is this the same as:

```suggestion
          unread = discussion && counts.reject { |(_, seen), _| seen }.present?
```

?

Reply:
Good shout but no (I had to try it to realise this myself). If discussion is `nil` then the expression returns `nil` not `false`.

Reply:
Okay!